{"name": "FireBase_XSS injection attempt in finding ID \"<script>alert('XSS');</script>\"", "status": "broken", "statusDetails": {"message": "requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyD08yxn7V_YAfkAY8mKDTYjfKV_qoPrdRQ\n", "trace": "  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\model.py\", line 1991, in run\n    match.run(runner.context)\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\matchers.py\", line 105, in run\n    self.func(context, *args, **kwargs)\n  File \"features\\steps\\FireBase_get_a_specific_sca_by_id.py\", line 105, in step_impl\n    id_token = get_firebase_id_token()\n               ^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\utilities\\authentication.py\", line 16, in get_firebase_id_token\n    response.raise_for_status()  # Raise an error for non-2xx responses\n    ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\requests\\models.py\", line 1026, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\n"}, "steps": [{"name": "Given FireBase_Specific Scan by ID the user sends GET request with finding ID", "status": "broken", "statusDetails": {"message": "requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyD08yxn7V_YAfkAY8mKDTYjfKV_qoPrdRQ\n", "trace": "  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\model.py\", line 1991, in run\n    match.run(runner.context)\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\matchers.py\", line 105, in run\n    self.func(context, *args, **kwargs)\n  File \"features\\steps\\FireBase_get_a_specific_sca_by_id.py\", line 105, in step_impl\n    id_token = get_firebase_id_token()\n               ^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\utilities\\authentication.py\", line 16, in get_firebase_id_token\n    response.raise_for_status()  # Raise an error for non-2xx responses\n    ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\requests\\models.py\", line 1026, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\n"}, "start": 1757691385525, "stop": 1757691386026}, {"name": "When FireBase_the user receives the response", "status": "skipped", "start": 1757691386027, "stop": 1757691386027}, {"name": "Then FireBase_Specific Scan by ID the status code returned should be 404", "status": "skipped", "start": 1757691386027, "stop": 1757691386027}, {"name": "And FireBase_Specific Scan by ID the message should contain \"The current request is not defined by this API.\"", "status": "skipped", "start": 1757691386027, "stop": 1757691386027}], "start": 1757691385524, "stop": 1757691386027, "uuid": "2425bfd3-7424-47fc-8b06-bd3127b4f618", "historyId": "762fa83313557086bc3f9d6f30aefcdd", "fullName": "FireBase_Get Scan Details using Genr3d API: FireBase_XSS injection attempt in finding ID \"<script>alert('XSS');</script>\"", "labels": [{"name": "severity", "value": "normal"}, {"name": "tag", "value": "Security"}, {"name": "tag", "value": "Test"}, {"name": "feature", "value": "FireBase_Get Scan Details using Genr3d API"}, {"name": "framework", "value": "behave"}, {"name": "language", "value": "cpython3"}], "titlePath": ["features", "FireBase_Get Scan Details using Genr3d API"]}